### ISSUE TITLE
getCountQuery is not configurable for use with DISTINCT

### ISSUE THREAD

**Issue by maddin79** [spring-projects/spring-data-jpa#2509]
I need to do a distinct count with specification. The code in the SimpleJpaRepository for the getCountQuery method has a the following if: 
```
if (query.isDistinct()) {
    query.select(builder.countDistinct(root));
} else {
    query.select(builder.count(root));
}
```
But before that if the distinct of the query is never set, so I assume it is always false.  I also can not set the distinct in the specification for this query. Like this the if does not make any sense or do I miss something?

Thanks

Best 
Martin

**Comment by schauder** [comment]
What keeps you from calling `query.setDistinct(true)` in the specifications `toPredicate` method?

It is not a feature I'm particular fond of because Specifications really should focus on the predicate, but it does exist.

**Comment by gregturn** [comment]
@maddin79 Can you recheck things give we just merged #1380 (https://github.com/spring-projects/spring-data-jpa/commit/8ea70c1e420183c60df5cd3094def1c06a979148) to `main`. This should parse the `Distinct` keyword found in custom `count` methods, and thus ONLY apply the `distinct` inside the `select count(...)` function (not outside).

See if this resolves this resolves the issue for you.

**Comment by maddin79** [comment]
@schauder yes, you are right and in the meantime we found a different solution. We do not touch the query anymore in the specification itself. 

@gregturn As I can see your solution fixed the problem. Good to know for the future, that there is a way now.

Thanks for your help and work :).
